AVX-474: Add view modes to Pixy Viewer

dat-tran-matternet opened the pull request:
This change adds a menu option for the user to select three view modes:

  - Block:  Block (aka blob) detection mode
  - Video:  Video mode
  - Merge:  Merge of block and video mode

To support video mode, the handling of images transferred over USB is wrapped to
expose the images to Python.

Support for Windows is added by upgrading to Python3 and removing boost
dependency.

dat-tran-matternet committed:
AVX-474: Add view modes to Pixy Viewer

This change adds a menu option for the user to select three view modes:

  - Block:  Block (aka blob) detection mode
  - Video:  Video mode
  - Merge:  Merge of block and video mode

To support video mode, the handling of images transferred over USB is wrapped to
expose the images to Python.

Support for Windows is added by upgrading to Python3 and removing boost
dependency.

zalice commented on src/host/libpixyusb/src/pixyinterpreter.cpp:
```diff
@@ -162,7 +175,7 @@ void PixyInterpreter::interpreter_thread()
     chirp_access_mutex_.unlock();
-    usleep(15000); // wait for 15ms, ie give time for 
+    std::this_thread::sleep_for(std::chrono::milliseconds(15)); // wait for 15ms, ie give time for
   }
```
This is pre-existing: the comment is incomplete. Could you add words or remove the second half of it?

dat-tran-matternet replied:
done

rabad-matternet commented on scripts/build_libpixyusb_swig.bat:
```diff
@@ -0,0 +1,47 @@
+@ECHO off
```
Is this a Windows batch file?

dat-tran-matternet replied:
Correct. This is to support the Windows end-users.

rabad-matternet commented on src/host/libpixyusb/src/pixy.cpp:
```diff
@@ -90,7 +90,7 @@ extern "C"
 
-    if(return_value == 0) 
+    if(return_value == 0)
     {
```
It looks like we modified a third-party source file as our own now.  If it is indeed our own file, we should further modify to fit our coding style.

This comment applies to other files in this commit as well.

dat-tran-matternet replied:
done

dat-tran-matternet committed:
AVX-474: Decrease detection rate window size

Decrease detection rate window size to 10 seconds and update interval to 2 seconds

dat-tran-matternet committed:
AVX-474: Coding sytle

dat-tran-matternet committed:
AVX-474: Output block count to console
